src/app/shell/details/details-tabs/all-provider-workshops/all-provider-workshops.component.ts

```diff
@@ -47,17 +51,18 @@ export class AllProviderWorkshopsComponent implements OnInit, OnDestroy {
     this.destroy$.unsubscribe();
   }
 
+  getWorkshops(): void {
+    this.store.dispatch(new GetWorkshopsByProviderId(this.providerParameters));
+  }
+
   onPageChange(page: PaginationElement): void {
-    this.store
-      .dispatch(new OnPageChangeWorkshops(page))
-      .pipe(takeUntil(this.destroy$))
-      .subscribe(() => this.onGetWorkshops.emit());
+    this.currentPage = page;
+    Util.setPaginationParams(this.providerParameters, this.currentPage, this.providerParameters.size);
+    this.getWorkshops();
```

Comment:
same -> move     Util.setPaginationParams(this.providerParameters, this.currentPage, this.providerParameters.size);
to getWorkshops method, because it is a combination, you can pass nessessary item per page as param

my point - is to improve code! one of the rules of the good code approach is DRY - Dont Repeat Yourself